### PR TITLE
Adding signature verification support

### PR DESCRIPTION
**Pull request by Luzifer**

### CODE REVIEW

File: build-image/Dockerfile [hunk 1]
```diff
@@ -5,13 +5,17 @@ MAINTAINER Knut Ahlers <knut@ahlers.me>
...
     go get github.com/Luzifer/gobuilder/cmd/configreader
 
 ADD ./builder.sh /usr/bin/builder.sh
+ADD ./gpgkey.asc.enc /root/gpgkey.asc.enc
 
 RUN mkdir /root/.ssh && \
     echo "Host *\n\tStrictHostKeyChecking no\n" >> ~/.ssh/config && \
-    chmod 700 /root/.ssh
+    chmod 700 /root/.ssh && \
+    gpg --list-keys 2>&1 1>/dev/null && \
```
Comment (wking) [inline build-image/Dockerfile:17]: Is this just an “is my GnuPG installation busted” check?

File: build-image/Dockerfile [hunk 2]
```diff
@@ -5,13 +5,17 @@ MAINTAINER Knut Ahlers <knut@ahlers.me>
...
+    echo "keyserver-options auto-key-retrieve" >> ~/.gnupg/gpg.conf && \
+    sed -i "s/^keyserver .*$/keyserver hkp:\/\/keyserver.ubuntu.com/" ~/.gnupg/gpg.conf
```
Comment (wking) [inline build-image/Dockerfile:19]: Why Ubuntu's server and not the hkp://pool.sks-keyservers.net pool?

File: build-image/builder.sh [hunk 3]
```diff
@@ -43,6 +46,22 @@ if [ "$(cat /tmp/go-build/build_master)" == "${short_commit}" ]; then
   exit 130
 fi
 
+log "Verifying tag signatures..."
+for tag in ${tags}; do
+  if ( LANG=C git tag --verify ${tag} 2>&1 | grep "Good signature" ); then
+    LANG=C git tag --verify ${tag} 2>&1 | grep "gpg:" > /tmp/go-build/.signature_${tag}
```
Comment (wking) [inline build-image/builder.sh:52]: I was wondering how you were loading your web of trust, but it looks like [Git doesn't support that yet](http://mikegerwitz.com/papers/git-horror-story#automate).  In any case, it should really be the downloader's web of trust, so passing the signature back to them for them to verify it locally would let them ensure that gobuilder wasn't being fooled by an impersonator's signature (although they'd still have to trust gobuilder to be dealing honestly).

File: build-image/Dockerfile (hunk 1 above)
Comment (Luzifer) [inline build-image/Dockerfile:17]: This just initializes the `/root/` keyring and creates the configs which are modified below.

File: build-image/Dockerfile (hunk 2 above)
Comment (Luzifer) [inline build-image/Dockerfile:19]: My experience in the past was the ubuntu keyserver is the most stable while all others do have more or less outages breaking systems…

File: build-image/builder.sh (hunk 3 above)
Comment (wking) [inline build-image/builder.sh:52]: Ah, with the “grepping the raw GnuPG output” approach you _could_ use gobuilder WoT to check for impersonators.  But that makes it hard to use gobuilder for folks outside its WoT.  If you are auto-trusting in gobuilder, you should probably use the `%G?` format outlined [here](http://mikegerwitz.com/papers/git-horror-story#automate) to save yourself the trouble of keeping up with GnuPG's output (although that's likely pretty stable ;).

File: frontend/repository.html [hunk 4]
```diff
@@ -120,6 +150,23 @@
...
+          <div class="modal-dialog modal-lg">
+            <div class="modal-content">
+              <div class="modal-header">
+                <button type="button" class="close" data-dismiss="modal" aria-label="Close"><span aria-hidden="true">&times;</span></button>
+                <h4 class="modal-title" id="myModalLabel">Code-signature of this build</h4>
+              </div>
+              <div class="modal-body">
+                <pre>{{ signature }}</pre>
+                <p>
+                  For extra security you can <a href="/api/v1/{{repo}}/signed-hashes/{{branch}}">download</a> a signed list of checksums for this builds artifacts.
```
Comment (wking) [inline frontend/repository.html:164]: “builds” → “build's”.

Comment (Luzifer) [inline frontend/repository.html:164]: Thx. Fixed that typo in c807cab

**Pull request merged**